test/example/SnapshotGateTest.js

```diff
@@ -0,0 +1,717 @@
+          expect(offerDates.isValid()).is.true;
+          expect(offerDurations.isValid()).is.true;
+
+          // Create the offer
+          await offerHandler
+            .connect(operator)
+            .createOffer(offer, offerDates, offerDurations, disputeResolverId, agentId);
+          offers.push(offer);
+
+          // Required constructor params for Group
+          offerIds = [offerId];
+
+          // Create Condition
+          condition = mockCondition({
+            tokenAddress: snapshotGate.address,
+            threshold: "0",
+            maxCommits: tokenSupply,
+            tokenType: TokenType.NonFungibleToken,
+            tokenId: tokenId,
+            method: EvaluationMethod.SpecificToken,
+          });
+          expect(condition.isValid()).to.be.true;
+
+          // Create Group
+          group = new Group(groupId, seller.id, offerIds);
+          expect(group.isValid()).is.true;
+          await groupHandler.connect(operator).createGroup(group, condition);
+          groups.push(group);
+        }
+      }
+      // End of gated offers creation
+
+      // Create second seller offer
+      const mo = await mockOffer();
+      ({ offerDates, offerDurations } = mo);
+      offer = mo.offer;
+      offer.sellerId = "2"; // second seller
+      offer.price = "0";
+      offer.sellerDeposit = "0";
+      offer.quantityAvailable = "5";
+      offer.buyerCancelPenalty = "0";
+      disputeResolverId = "0";
+
+      // Check if entities are valid
+      expect(offer.isValid()).is.true;
+      expect(offerDates.isValid()).is.true;
+      expect(offerDurations.isValid()).is.true;
+
+      // Create the offer
+      let tx = await offerHandler
+        .connect(operator2)
+        .createOffer(offer, offerDates, offerDurations, disputeResolverId, agentId);
+      offers.push(offer);
+
+      const txReceipt = await tx.wait();
+      const event = getEvent(txReceipt, offerHandler, "OfferCreated");
+      otherSellerOfferId = event.offerId;
+    });
+
+    afterEach(async function () {
+      // Reset the accountId iterator
+      accountId.next(true);
+    });
+
+    context("👉 appendToSnapshot()", async function () {
+      it("should emit a SnapshotAppended event", async function () {
+        // Batch of one
+        const batch = snapshot.slice(0, 1);
+
+        await expect(snapshotGate.connect(deployer).appendToSnapshot(batch)).to.emit(snapshotGate, "SnapshotAppended");
+      });
+
+      it("should allow multiple invocations", async function () {
+        // Split snapshot into two batches
+        const batchSize = Math.floor(snapshot.length / 2);
+        const batch1 = snapshot.slice(0, batchSize);
+        const batch2 = snapshot.slice(batchSize + 1);
+
+        await expect(snapshotGate.connect(deployer).appendToSnapshot(batch1)).to.emit(snapshotGate, "SnapshotAppended");
+
+        await expect(snapshotGate.connect(deployer).appendToSnapshot(batch2)).to.emit(snapshotGate, "SnapshotAppended");
+      });
```

Comment:
Check event args. Check event args. We have some examples of checking array args for DisputeResolverCreated.

Reply:
fixed